### ISSUE TITLE
Buffer delete : error - Key not present in Dictionary: 5

### ISSUE THREAD

**Issue by g1ra** [weynhamz/vim-plugin-minibufexpl#18]
Frequently I got an error message when deleting a buffer: 

> Error detected while processing function <SNR>74_BufDeleteHandler..<SNR>74_StartExplorer..<SNR>74_BuildAllBufferDicts..<SNR>74_BuildBufferFinalDict..<SNR>74_BuildBufferPathSignDict:  
> line   26:  
> E716: Key not present in Dictionary: 5  
> E116: Invalid arguments for function len(s:bufPathDict[bufnr]) < index  
> E15: Invalid expression: len(s:bufPathDict[bufnr]) < index  
> line   34:  
> E716: Key not present in Dictionary: 5  
> E116: Invalid arguments for function get(s:bufPathDict[bufnr],index)  
> E15: Invalid expression: get(s:bufPathDict[bufnr],index)  
> line   36:  
> E121: Undefined variable: part  
> E116: Invalid arguments for function empty(part)  
> E15: Invalid expression: empty(part)  
> line   41:  
> E121: Undefined variable: part  
> E116: Invalid arguments for function has_key(partDict, part))  
> E15: Invalid expression: (!has_key(partDict, part))  
> line   44:  
> E121: Undefined variable: part  
> E116: Invalid arguments for function add  
> Error detected while processing function <SNR>74_BufDeleteHandler..<SNR>74_StartExplorer..<SNR>74_BuildAllBufferDicts..<SNR>74_BuildBufferFinalDict..<SNR>74_BuildBufferPathSignDict..<SNR>74_BuildBufferPathSignDict:  
> line   26:  
> E716: Key not present in Dictionary: 5  
> E116: Invalid arguments for function len(s:bufPathDict[bufnr]) < index  
> E15: Invalid expression: len(s:bufPathDict[bufnr]) < index  
> line   34:  
> E716: Key not present in Dictionary: 5  
> E116: Invalid arguments for function get(s:bufPathDict[bufnr],index)  
> E15: Invalid expression: get(s:bufPathDict[bufnr],index)  
> line   36:  
> E121: Undefined variable: part  
> E116: Invalid arguments for function empty(part)  
> E15: Invalid expression: empty(part)  
> line   41:  
> E121: Undefined variable: part  
> E116: Invalid arguments for function has_key(partDict, part))  
> E15: Invalid expression: (!has_key(partDict, part))  
> line   44:  
> E121: Undefined variable: part  
> E116: Invalid arguments for function add  
> Error detected while processing function <SNR>74_BufDeleteHandler..<SNR>74_StartExplorer..<SNR>74_BuildAllBufferDicts..<SNR>74_BuildBufferFinalDict..<SNR>74_BuildBufferPathSignDict..<SNR>74_BuildBufferPathSignDict..<SNR>74_BuildBu
> fferPathSignDict:  
> line   26:  
> E716: Key not present in Dictionary: 5  
> E116: Invalid arguments for function len(s:bufPathDict[bufnr]) < index  
> E15: Invalid expression: len(s:bufPathDict[bufnr]) < index  
> line   34:  
> E716: Key not present in Dictionary: 5  
> E116: Invalid arguments for function get(s:bufPathDict[bufnr],index)  
> E15: Invalid expression: get(s:bufPathDict[bufnr],index)  
> line   36:  
> E121: Undefined variable: part  
> E116: Invalid arguments for function empty(part)  
> E15: Invalid expression: empty(part)  
> line   41:  
> E121: Undefined variable: part  
> E116: Invalid arguments for function has_key(partDict, part))  
> E15: Invalid expression: (!has_key(partDict, part))  
> line   44:  
> E121: Undefined variable: part  
> E116: Invalid arguments for function add  
> Error detected while processing function <SNR>74_BufDeleteHandler..<SNR>74_StartExplorer..<SNR>74_BuildAllBufferDicts..<SNR>74_BuildBufferFinalDict..<SNR>74_BuildBufferPathSignDict..<SNR>74_BuildBufferPathSignDict..<SNR>74_BuildBu
> fferPathSignDict..<SNR>74_BuildBufferPathSignDict:  
> line   26:  
> E716: Key not present in Dictionary: 5

**Comment by weynhamz** [comment]
Yes, I came across this problem several times when work with Vim sesssions, I am investigating, will keep you post.

**Comment by g1ra** [comment]
Thank you very much !
